### DISCUSSION TITLE
Unclear how to access array elements in `{{< meta >}}`

### DISCUSSION THREAD

**Discussion by multimeric** [quarto-dev/quarto-cli#6764] [Q&A] [answered]
### Description

The quarto documentation explains how to use metadata within the document, and also how to access nested values using the dot notation: https://quarto.org/docs/authoring/variables.html. However, it doesn't explain how to access array/list elements, and indeed I can't find a syntax that supports this. 

In the following document I show my attempts to do this:

```
---
top: "Top Lvel"
nested:
    str: "Nested String"
    array:
        - foo
        - bar
---

{{< meta top >}}
{{< meta nested.str >}}
{{< meta nested.array[0] >}}
{{< meta nested.array[1] >}}
{{< meta nested.array.0 >}}
{{< meta nested.array.1 >}}
```

This renders to PDF like this, showing that none of my attempted array accesses worked:

![Screen Shot 2023-09-08 at 11 57 41 am](https://github.com/quarto-dev/quarto-cli/assets/5019367/310681ed-7452-4002-8914-51ddccb5c8f4)

**Comment by cderv** [comment]
It is currently a limitation of how shortcodes works - it does not support yet arrays or maps. 

It is not mentioned in the doc, but I can see reference of this limitation in our code comment
https://github.com/quarto-dev/quarto-cli/blob/a67ae2a0a17b855d06deff93c156836e08061441/src/resources/filters/quarto-pre/shortcodes-handlers.lua#L126-L130

I'll open an issue to track this as possible improvement. cc @dragonstyle

**Reply by cderv** [reply]
See 
* #6769

**Comment by cderv** [comment, marked as answer]
This will be possible in 1.4 now using the following syntax 

````markdown
---
top: "Top Lvel"
nested:
    str: "Nested String"
    array:
        - foo
        - bar
---

{{< meta top >}}
{{< meta nested.str >}}
{{< meta nested.array.1 >}}
````

It will be available from v1.4.535
